torch_em/model/unetr.py

```diff
@@ -408,11 +410,337 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         return x
 
 
+class UNETR2D(UNETR):
+    """A two-dimensional UNet Transformer using a vision transformer as encoder and a convolutional decoder.
+    """
+    pass
+
+
+class UNETR3D(nn.Module):
+    """A three dimensional UNet Transformer using a vision transformer as encoder and a convolutional decoder.
+    """
+    def __init__(
+        self,
+        model_type: str,
+        backbone: Literal["sam", "sam2", "mae", "scalemae", "dinov3"] = "sam2",
+        out_channels: int = 3,
+        img_size: int = 1024,
+        final_activation: Optional[Union[str, nn.Module]] = None,
+        checkpoint_path: Optional[Union[str, os.PathLike]] = None,
+        use_strip_pooling: bool = True,
+    ):
+        super().__init__()
+
+        self.img_size = img_size
+
+        # Step 1: the image encoder backbone.
+        image_encoder = get_vision_transformer(backbone=backbone, model=model_type, img_size=img_size)
+
+        # Load the pretrained image encoder weights
+        if checkpoint_path is not None:
+            state = torch.load(checkpoint_path, map_location="cpu")["model"]
```

Comment:
I wouldn't hard-code the save key as `"model"`. I think that we need a single logic as https://github.com/constantinpape/torch-em/blob/main/torch_em/model/unetr.py#L49 to handle this for different models.